lib/web/integrations_awsoidc.go

```diff
@@ -475,6 +475,11 @@ func (h *Handler) awsOIDCConfigureIdP(w http.ResponseWriter, r *http.Request, p
 		return nil, trace.BadParameter("invalid role %q", role)
 	}
 
+	proxyAddr := h.PublicProxyAddr()
+	if !strings.HasPrefix(proxyAddr, "https://") && !strings.HasPrefix(proxyAddr, "http://") {
+		proxyAddr = "https://" + proxyAddr
+	}
+
```

Comment:
Can we re-use `lib/integrations/awsoidc/issuer.go` `IssuerFromPublicAddress` instead?
It does some extra validation but I still think it's worth to re-use it here